packages/babel-traverse/src/scope/index.ts

```diff
@@ -899,6 +870,21 @@ export default class Scope {
     };
 
     this.crawling = true;
+    // traverse does not visit the root node, here we explicitly collect
+    // root node binding info when the root is not a Program.
+    if (programParent.path !== path && collectorVisitor._exploded) {
```

Comment:
`programParent.path !== path` is meant to to avoid visiting logic when the path is a program. The program crawl is always performed before `babel.transform` so it is very popular path.

Reply:
Can we just use `path.type !== "Program"` instead?